File: client/common/protocolo/protocoloRespuesta.go
package protocolo

import (
	"net"

	"github.com/7574-sistemas-distribuidos/docker-compose-init/client/common/modelo"
	"github.com/7574-sistemas-distribuidos/docker-compose-init/client/common/serializacion"
)

func RecibirRespuesta(conn net.Conn) (*modelo.Respuesta, error) {
	texto, err := recibirString(conn)
	if err != nil {
		return nil, err
	}

	return serializacion.RespuestaDesdeString(texto), nil
}

File: server/common/protocolo/protocolo_respuesta.py
import socket

from common.modelo.respuesta import Respuesta
from common.protocolo.protocolo_base import enviar_string
from common.serializacion.serializacion_respuesta import respuesta_a_string


def enviar_respuesta(socket: socket.socket, respuesta: Respuesta) -> int:
    respuesta_str = respuesta_a_string(respuesta)
    return enviar_string(socket, respuesta_str)